### PR TITLE
Foundry and Docker Section

### PR DESCRIPTION
**Pull request by dmfxyz**
This is a write-up of the Docker installation instructions and a walkthrough of how to use foundry within docker. It covers:

- Basic Installation
- Direct Interfacing w/ `forge` and `cast` through Docker
- Live developing in Docker using `--watch`
- Building custom images on top of the Foundry image

There will likely be more to add later, but this covers a good first portion.

### CODE REVIEW

File: src/tutorials/foundry-docker.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+Once installed, you can download the latest release by running:  
+ `docker pull ghcr.io/gakonst/foundry:latest`
+
+ To keep future commands succint, let's re-tag the image:  
+ `docker tag ghcr.io/gakonst/foundry:latest foundry:latest`
+
+Having forge/cast installed locally is not strictly required, but may be helpful for debugging. You can install them using [foundryup](/getting-started/installation.html#using-foundryup).
+
+Finally, to use any of the `cast` portions of this tutorial, you will need access to an Ethereum node. If you don't have your own node running (likely), you can use a Node as a service. We won't recommend a specific provider in this tutorial. A good place to start learning about Nodes-as-a-Service is [Ethereum's article](https://ethereum.org/en/developers/docs/nodes-and-clients/nodes-as-a-service/) on the subject.
```
Comment (dmfxyz) [inline src/tutorials/foundry-docker.md:21]: Also to use `create`

File: src/tutorials/foundry-docker.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
+# Dockerizing a Foundry project
```
Comment (onbjerg) [inline src/tutorials/foundry-docker.md:1]: we use ## for top-level in pages

File: src/tutorials/foundry-docker.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+This tutorial shows you how to build, test, and deploy a smart contract using Foundry's Docker image. It adapts code from the [solmate nft](./solmate-nft.md) tutorial. If you haven't completed that tutorial yet, and are new to solidity, you may want to start with it first. Alternatively, if you have some familiarity with Docker and Solidity, you can use your own existing project and adjust accordingly. The full source code for both the NFT and the Docker stuff is available [here](https://github.com/dmfxyz/foundry-docker-tutorial).
+
+#####  This tutorial is for illustrative purposes only and provided on an as-is basis. The tutorial is not audited nor fully tested. No code in this tutorial should be used in a production environment.
+
+
```
Comment (onbjerg) [inline src/tutorials/foundry-docker.md:7]:
```suggestion

```

File: src/getting-started/installation.md [hunk 4]
```diff
@@ -41,3 +41,28 @@ After this, run the following to build Foundry from source:
...
+Build, test, fuzz, formally verify, debug & deploy solidity contracts.
+
+USAGE:
+    forge <SUBCOMMAND>
+
+OPTIONS:
+    -h, --help       Print help information
+    -V, --version    Print version information
+...
+```
```
Comment (onbjerg) [inline src/getting-started/installation.md:63]: seems like this is duped

File: src/getting-started/installation.md [hunk 5]
```diff
@@ -41,3 +41,28 @@ After this, run the following to build Foundry from source:
...
+It is also possible to build the docker image locally. From the Foundry repository, run:
+```sh
+docker build -t foundry .
+```
+##### Note: Some machines (including those with M1 chips) may be unable to build the docker image locally. This is a known issue.
```
Comment (onbjerg) [inline src/getting-started/installation.md:68]: We should use the > notes that we use other places in the book

**Review by onbjerg — COMMENTED** [review]
otherwise lgtm so far :)

File: src/getting-started/installation.md (hunk 4 above)
Comment (dmfxyz) [reply src/getting-started/installation.md]: agreed, simplified to just the pull command. More detail stuffed is in the docker section.

**Comment by dmfxyz** [comment]
Thanks @onbjerg! Addressed it all and made a few more editorial changes. Moving out of draft mode.

**Review by onbjerg — APPROVED** [review]
LGTM! Thanks!!

**Comment by onbjerg** [comment]
@all-contributors add @dmfxyz for docs

**Pull request merged**